2018/2018_File_Generation_Features.md

```diff
@@ -0,0 +1,67 @@
+# 2018 File Generation Features
+
+## The FileGenerator Class. 
+The `FileGenerator()` class is located in [`file_generator.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/file_generator.py) and instantiates without any inputs in the function call. The class uses data stored with yaml configuration files in the 
+[`python/configurations/`](https://github.com/cfpb/hmda-test-files/tree/master/2018/python/configurations) directory. The following yaml configuration files can be altered to create HMDA submission test files.  
+
+-[`test_filepaths.yaml`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/configurations/test_filepaths.yaml) holds the filepath names for storing clean files, edit test files, log files, and edit reports as well as the paths to the TS and LAR schemas. 
+
+-[`geography_data.yaml`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/configurations/geographic_data.yaml) holds the filepaths for geographic dependency data such as the tract to CBSA file. The configuration includes the column names for the CBSA file and the columns used for file creation. The file also contains a path to a list of zip codes and a dictionary of FIPS state codes. 
+
+-[`clean_file_config.yaml`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/configurations/clean_file_config.yaml) holds values for the test institution, file length, and file name for generating a clean HMDA submission file for testing. Other variables, such as data ranges can also be set in this file. 
+
+Clean files will pass syntax and validity edits. Edit test files will fail at least one (and sometimes more than one) edit. These files will be used as a double check for the implementation of the HMDA edits for the 2018 HMDA Platform. Creating edit test files is a two step process. The generation of edit test files requires a clean data file to be present. If edit files already exist with the same names as those produced by this script, they will be overwritten.
+
+Edit test files are written to the directory indicated in `test_filepaths.yaml` based on edit type, such as /edits_files/test_files/Bank1/syntax/s300.txt. The file name indicates which edit the file is designed to test. If an edit has multiple conditions, a file will be made for each condition in the format /edits_files/test_files/Bank1/syntax/s301_1.txt.
+
+### Changing Edit Logic and the Classes Used for File Creation 
+The FileGenerator Class relies on four other classes, each with its own role in file creation. 
+
+-`lar_generator` in [`lar_generator.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/lar_generator.py) creates raw TS and LAR rows, the values of which are then manipulated to create clean test files or edit test files. 
```

Comment:
After `creates raw TS and LAR rows` can you add something like "using the TS and LAR schema.json files" with appropriate links?

This integrates more of the code in the user facing documentation and helps explain how code files relate to each other.

Reply:
Sounds good! This change has been implemented.